src/WarcraftLegacies.Source/Quests/Warsong/QuestScepterOfTheQueenWarsong.cs

```diff
@@ -0,0 +1,35 @@
+﻿using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.QuestSystem;
+using MacroTools.QuestSystem.UtilityStructs;
+using WarcraftLegacies.Source.Setup;
+using WarcraftLegacies.Source.Setup.Legends;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace WarcraftLegacies.Source.Quests.Warsong
+{
+    public class QuestScepterOfTheQueenWarsong : QuestData
+    {
+        public QuestScepterOfTheQueenWarsong(Rectangle area) : base("Royal Plunder", "Remnants of the ancient Highborne survive within the ruins of Dire Maul. If Feathermoon Stronghold falls, it would become a simple matter to slaughter the Highborne and plunder their artifacts.", "ReplaceableTextures\\CommandButtons\\BTNNagaWeaponUp2.blp")
+        {
+            HighBourneArea = area;
+            AddObjective(new ObjectiveLegendNotPermanentlyDead(LegendWarsong.StonemaulKeep));
+            AddObjective(new ObjectiveLegendDead(LegendSentinels.legendFeathermoon));
+            AddObjective(new ObjectiveAnyUnitInRect(HighBourneArea, "Dire Maul", true));
+            HighBourneArea.Rect.MakeUnitsInvulnerable();
```

Comment:
So I think this isn't quite the pattern we want. We should:
1. When the Quest is created, make all the units invulnerable and put them into a List<unit> field.
2. When the Quest is completed, iterate across that list and rescue the units inside.
3. When the Quest is failed, iterate across that list and rescue the units for neutral hostile.

The reason for this is so that we can avoid accidentally rescuing neutral passive units that may have been moved to the region after the game has already started. We don't want to rescue those.

Reply:
Sorry, scratch point 3. For this quest we can't give it to hostile on fail, since the rival player might still complete their version of the quest.

Reply:
To Addresss this, `MakeUnitsInvulnerable` now returns the units found.
And then I added another Method to `PlayerExtensions` that takes a List of units but does basically the same as 
`ChangeUnitsOwngingPlayer` in rect extensions.
I did this because the rect is no longer needed there